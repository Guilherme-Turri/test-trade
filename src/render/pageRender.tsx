import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../styles/theme';
import store from '../storage/store';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

export const pageRender = (children: React.ReactNode) => {
  return render(
    <ThemeProvider theme={theme}>
      <Provider store={store}>
        <BrowserRouter>
          {children}
        </BrowserRouter>
      </Provider>
    </ThemeProvider>);
};