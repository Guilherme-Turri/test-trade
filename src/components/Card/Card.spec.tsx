import { componentRender } from '../../render/componentRender'
import { screen, fireEvent } from '@testing-library/react'
import { Card } from './'

describe('</Card>', () => {
  const cardProps = {
    title: 'cardTitle',
    img: <img src="test.jpg" alt="Test" />,
    handleOpen: jest.fn(),
    available: true,
  }
  it('should render properly and avaliable to click', () => {
    componentRender(<Card {...cardProps} />)
    const title = screen.getByText('cardTitle');
    const alt = screen.queryByAltText('Test');
    const card = screen.getByTestId('cardId')
    expect(title).toBeInTheDocument();
    expect(alt).toBeInTheDocument();
    fireEvent.click(card);
    expect(cardProps.handleOpen).toHaveBeenCalled();;
  })
  it('should render properly and not avaliable to click', () => {
    cardProps.available = false;
    const { container } = componentRender(<Card {...cardProps} />)
    const title = screen.getByText('cardTitle');
    const alt = screen.queryByAltText('Test');
    const card = screen.getByTestId('cardId')
    expect(title).toBeInTheDocument();
    expect(alt).toBeInTheDocument();
    fireEvent.click(card);
    expect(cardProps.handleOpen).not.toHaveBeenCalled();
    expect(container).toMatchSnapshot();
  })
})
