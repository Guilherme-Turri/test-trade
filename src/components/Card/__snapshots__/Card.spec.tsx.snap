// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`</Card> should render properly and not avaliable to click 1`] = `
.c3 {
  color: #2d5839;
}

.c0 {
  background: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 15.625rem;
  width: 15.625rem;
  border-radius: 5px;
  margin: 0.938rem;
  box-shadow: none;
}

.c0 .c2 {
  color: grey;
  font-family: 'Bruno Ace SC',cursive;
  border-bottom: none;
  padding: 0.313rem;
}

.c1 {
  height: 6.25rem;
  width: 6.25rem;
}

<div>
  <div
    class="c0"
    data-testid="cardId"
  >
    <div
      class="c1"
    >
      <img
        alt="Test"
        src="test.jpg"
      />
    </div>
    <p
      class="c2 c3"
    >
      cardTitle
    </p>
  </div>
</div>
`;
